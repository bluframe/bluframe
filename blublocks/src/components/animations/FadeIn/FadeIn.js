// @flow

import { type ComponentProps } from "."
import React from "react"
import { Wrapper } from "./styled"

const FadeIn = ({
  children,
  fadeIn,
  isAnimated
}: ComponentProps): React$Node => (
  <Wrapper $fadeIn={fadeIn} $isAnimated={isAnimated}>
    {children}
  </Wrapper>
)

export default FadeIn
