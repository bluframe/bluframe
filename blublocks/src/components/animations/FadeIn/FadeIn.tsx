import { ComponentProps } from "."
import React from "react"
import { Wrapper } from "./styled"

const FadeIn = ({ children, fadeIn, isAnimated }: ComponentProps) => (
  <Wrapper $fadeIn={fadeIn} $isAnimated={isAnimated}>
    {children}
  </Wrapper>
)

export default FadeIn
