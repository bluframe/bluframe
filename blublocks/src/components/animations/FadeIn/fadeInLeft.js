// @flow

import { type KeyFrames, keyframes } from "styled-components"

const fadeInLeft: KeyFrames = keyframes`
0% {
    opacity: 0;
    transform: translateX(-20px);
  }

  100% {
    opacity: 1;
    transform: translateX(0);
  }
`

export default fadeInLeft
