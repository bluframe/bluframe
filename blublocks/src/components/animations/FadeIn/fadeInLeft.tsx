import { keyframes } from "styled-components"

const fadeInLeft = keyframes`
0% {
    opacity: 0;
    transform: translateX(-20px);
  }

  100% {
    opacity: 1;
    transform: translateX(0);
  }
`

export default fadeInLeft
