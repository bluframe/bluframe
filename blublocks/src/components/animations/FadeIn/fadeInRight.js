// @flow

import { type KeyFrames, keyframes } from "styled-components"

const fadeInRight: KeyFrames = keyframes`
  0% {
    opacity: 0;
    transform: translateX(20px);
  }

  100% {
    opacity: 1;
    transform: translateX(0);
  }
`

export default fadeInRight
