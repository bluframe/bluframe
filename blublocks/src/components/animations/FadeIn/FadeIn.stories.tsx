import FadeIn, { Props } from "components/animations/FadeIn"
import React from "react"

const stories = {
  component: FadeIn,
  title: "Animations/FadeIn"
}

const Template = (args: Props) => <FadeIn {...args}>{"Hello!"}</FadeIn>

export const Default: any = Template.bind({})
Default.args = {
  fadeIn: "Left",
  isAnimated: false
}

export const Left: any = Template.bind({})
Left.args = {
  fadeIn: "Left",
  isAnimated: true
}

export const Right: any = Template.bind({})
Right.args = {
  fadeIn: "Right",
  isAnimated: true
}

export const Up: any = Template.bind({})
Up.args = {
  fadeIn: "Up",
  isAnimated: true
}

export default stories
