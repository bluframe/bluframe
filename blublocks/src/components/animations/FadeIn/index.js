// @flow

import FadeIn from "./FadeIn"

export type Props = {|
  +children: React$Node,
  +fadeIn: "Left" | "Right" | "Up",
  +isAnimated: boolean
|}

export type ComponentProps = {|
  +children: React$Node,
  +fadeIn: "Left" | "Right" | "Up",
  +isAnimated: boolean
|}

export default FadeIn
