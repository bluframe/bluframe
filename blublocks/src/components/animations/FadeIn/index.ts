import FadeIn from "./FadeIn"
import React from "react"

export interface Props {
  children: React.ReactNode
  fadeIn: "Left" | "Right" | "Up"
  isAnimated: boolean
}

export interface ComponentProps extends Props {}

export default FadeIn
