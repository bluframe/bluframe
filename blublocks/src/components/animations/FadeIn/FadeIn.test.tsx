import FadeIn from "./FadeIn"
import { ComponentProps } from "."
import React from "react"
import { render } from "tests"

describe("FadeIn", () => {
  it("renders", () => {
    const props: Omit<ComponentProps, "children"> = {
      fadeIn: "Left",
      isAnimated: false
    }

    const { container } = render(
      <FadeIn {...props}>
        <div />
      </FadeIn>
    )

    expect(container.firstChild).toMatchSnapshot()
  })
})
