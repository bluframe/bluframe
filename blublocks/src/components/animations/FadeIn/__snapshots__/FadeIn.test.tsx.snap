// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FadeIn renders 1`] = `
<div
  class="sc-bdDqHD fgEUFx"
>
  <div />
</div>
`;
